components/db/testimony/usePublishedTestimonyListing.ts

```diff
@@ -91,7 +100,21 @@ async function listTestimony(
   limitCount: number,
   startAfterKey: unknown | null
 ): Promise<Testimony[]> {
+  if (refinement.profilePage && refinement.uid) {
```

Comment:
we should move the whole hook over to the typesense client and handle pagination and refinements